sql/core/src/test/scala/org/apache/spark/sql/execution/ui/MetricsAggregationBenchmark.scala

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.ui
+
+import java.util.Properties
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.collection.mutable
+import scala.concurrent.duration._
+
+import org.apache.spark.{SparkConf, TaskState}
+import org.apache.spark.benchmark.{Benchmark, BenchmarkBase}
+import org.apache.spark.executor.ExecutorMetrics
+import org.apache.spark.internal.config.Status._
+import org.apache.spark.scheduler._
+import org.apache.spark.sql.execution._
+import org.apache.spark.sql.execution.metric.SQLMetricInfo
+import org.apache.spark.status.ElementTrackingStore
+import org.apache.spark.util.{AccumulatorMetadata, LongAccumulator, Utils}
+import org.apache.spark.util.kvstore.InMemoryStore
+
+object MetricsAggregationBenchmark extends BenchmarkBase {
```

Comment:
Could you add the generated result(JDK8/JDK11) into this PR, please?

Reply:
You mean add a comment in the code? What is the point?

If you modify the code again, you have to run the benchmark both with the old code and the new code and post your results. The results on my machine will be different than on anyone else's.

And JDK8/JDK11 differences are kinda irrelevant here.

Reply:
@vanzin . No~ What you need is just run the benchmark with `SPARK_GENERATE_BENCHMARK_FILES=1`. It will generate the result for you.
> You mean add a comment in the code? What is the point?

Did you take a look at the other benchmarks recently? For example, the following.
```scala
/**
 * Benchmark for CoalescedRDD.
 * Measures rdd.coalesce performance under various combinations of
 * coalesced partitions and preferred hosts
 * To run this benchmark:
 * {{{
 *   1. without sbt:
 *      bin/spark-submit --class <this class> --jars <spark core test jar>
 *   2. build/sbt "core/test:runMain <this class>"
 *   3. generate result:
 *      SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "core/test:runMain <this class>"
 *      Results will be written to "benchmarks/CoalescedRDD-results.txt".
 * }}}
 * */
```

Since Apache Spark 3.0.0, we are tracking all the benchmark results with the generated files between Apache Spark versions.

cc @cloud-fan

Reply:
I had not seen that. Ok, I'll add the results file even though this is kind of an odd benchmark.

Reply:
Thank you, @vanzin !

Reply:
I added the files but it's still rather pointless, in my view. A single measurement at some random point in time made with some random hardware is not useful for comparison.

Whoever updates the code still needs to redo those measurements with both the old an new version, because the values in those files don't mean anything.

And also you'd better hope those instructions copy & pasted into dozens of files never change.

Reply:
I completely agree with you, @vanzin ! I also have been noticing that those different and mixed results exists everywhere (especially Mac/Server results.)

> A single measurement at some random point in time made with some random hardware is not useful for comparison.

That's the reason we recommend `r3.xlarge` EC2 instance as a standard machine for a long time and try to help to generate the result on that. All the existing one is the `r3.xlarge` generated result. Additionally, I will do the regeneration before 3.0.0 release to make a baseline.

Also, the current `BenchmarkBase` is a way for us to automate in the downstream. We can track the performance by commit by commit (if you want).

Reply:
You don't need those files to automate things. You start from somewhere to define a baseline, then compare from that. The results don't need to be committed to git. They can be managed by that automation.

The process you mention also ignores changes to the benchmarks themselves, even if those should be rarer.

But in any case, I just want to get this code in and don't really care that much about what you're doing with these files. As I said, this benchmark is different from the others and the base class was just a way to reuse code for measurements.